test/e2e/node_reuse.go

```diff
@@ -103,9 +120,17 @@ func nodeReuse(ctx context.Context, inputGetter func() NodeReuseInput) {
 	kcpObj.Spec.MachineTemplate.InfrastructureRef.Name = newM3machineTemplateName
 	kcpObj.Spec.Version = upgradedK8sVersion
 	kcpObj.Spec.RolloutStrategy.RollingUpdate.MaxSurge.IntVal = 0
-	Expect(helper.Patch(ctx, kcpObj)).To(Succeed())
+	Logf("Disable tainting of CP nodes during the remainder of the node reuse test [node_reuse]")
+	kcpObj.Spec.KubeadmConfigSpec.JoinConfiguration.NodeRegistration.Taints = emptyTaint
+	kcpObj.Spec.KubeadmConfigSpec.InitConfiguration.NodeRegistration.Taints = emptyTaint
+	patchRequestError := helper.Patch(ctx, kcpObj)
+	if patchRequestError != nil {
+		Logf("Error while patching KCP: %s [node_reuse]", patchRequestError.Error())
+	}
+	Expect(patchRequestError).NotTo(HaveOccurred())
```

Comment:
This seems a bit unnecessary. Is it just to get an explicit log message out if there was an error? If there was an error the `Expect` would anyway show it.

Reply:
That is the thing, it was failing without a log, it could have been for an other reason but the expect was failing and there was no log on multiple tests, which was strange...

Reply:
Weird. Maybe it would need a comment passed to `NotTo`. But anyway, let's keep it like this for now.